core/services/ocr2/plugins/rebalancer/plugin.go

```diff
@@ -400,6 +404,48 @@ func (p *Plugin) computePendingTransfersConsensus(observations []models.Observat
 	return quorumEvents, nil
 }
 
+// computeMedianGraph computes a graph with the provided median liquidities per chain.
+// The nodes in the resulting graph are copied from the plugin node. Meaning that different
+// plugin instances might have different graph.
+func (p *Plugin) computeMedianGraph(medianLiquidities []models.NetworkLiquidity) (liquiditygraph.LiquidityGraph, error) {
+	g := liquiditygraph.NewGraph()
+
+	for _, medianLiq := range medianLiquidities {
+		sourceNetwork := medianLiq.Network
+
+		neighbors, exists := p.liquidityGraph.GetNeighbors(sourceNetwork)
+		if !exists {
+			// this node is not aware of the network
+			continue
+		}
+
+		switch g.HasNetwork(sourceNetwork) {
+		case true: // was added while processing a source network
```

Comment:
Clearer comment - I think you mean here that this was processed as another network's neighbor right?

```suggestion
		case true: // was added while processing another source network
```